### ISSUE TITLE
Eng-hindi transformer translation model

### ISSUE THREAD

**Issue by aj7tesh** [pytorch/fairseq#1681]
I am trying to train a NMT transformer model from english to hindi on a corpus of around 2M. But my training is not at all converging. My ppl and loss are getting stagnant in the first epoch it self. I have gone through several experiments but unable to decode the issue. Below are the hyperparameters I am using.
CUDA_VISIBLE_DEVICES=0 python train.py data-bin/english_hindi --lr 0.005  --dropout 0.2 --max-tokens 2048 --arch transformer --save-dir checkpoints/models_anv --validate-interval 1 --max-epoch 10 --save-interval 1  --optimizer adam --adam-betas '(0.9, 0.98)' --warmup-updates 4000 --label-smoothing 0.1 --criterion label_smoothed_cross_entropy --weight-decay 0.0 --log-format tqdm --log-interval=50 --clip-norm 0.0 --lr-scheduler inverse_sqrt --warmup-init-lr 1e-07

I have tried changling lr , batch size etc. but no help

epoch 001:   3549 / 29783 loss=10.719, nll_loss=9.852, ppl=924.139, wps=11130.8, ups=6.43, wpb=1731.3, bsz=66.1, num_updates=3550, lr=0.00443751, gnorm=1.155, clip=0,

this is where it is stuck with very little up and down

**Comment by myleott** [comment]
That learning rate is probably way too high. Try reducing to 0.0005.

**Comment by aj7tesh** [comment]
thanks @myleott  I will try and let you know.
Also how do you build a server around this model. Not clear documentation for fairseq unlike OpenNMT.
I am externally tokenizing and encoding my training sentences and then putting them to train. Once model is trained how do i test it other than interactive.py and generate.py

**Comment by lematt1991** [comment]
I would recommend using the [torch hub](https://github.com/pytorch/fairseq/tree/master/examples/translation#example-usage-torchhub) if you want to test without using `interactive.py` or `generate.py`

**Comment by lematt1991** [comment]
Seems this has been resolved.  Please open a new issue if you are still having problems.

**Comment by Dornavineeth** [comment]
> thanks @myleott I will try and let you know.
> Also how do you build a server around this model. Not clear documentation for fairseq unlike OpenNMT.
> I am externally tokenizing and encoding my training sentences and then putting them to train. Once model is trained how do i test it other than interactive.py and generate.py

@aj7tesh can you please elaborate and point to any source for externally tokenizing and encoding training sentences (eng,hindi)
